include/oneapi/dpl/experimental/kt/esimd_radix_sort_cooperative.h

```diff
@@ -392,9 +394,13 @@ void cooperative(sycl::queue __q, _Range&& __rng, ::std::size_t __n) {
             KeyT, RADIX_BITS, THREAD_PER_TG, /*PROCESS_SIZE*/ 256, IsAscending, _EsimRadixSort>()(
                 __q, ::std::forward<_Range>(__rng), __n, __groups, __tmpbuf, p_sync);
     }
+
+    // TODO: Required to remove this wait
     __e.wait();
```

Comment:
@dmitriy-sobolev we have problem here: required to call ```sycl::free``` after finish event waiting.
1) Should we try to return ```__future``` instance and try to pack ```sycl::free``` call into ```__future```  destructor?
2) or this ```wait()``` call is OK at current moment here while we going to implement memory allocation outside of this ```radix_sort``` algorithm...